### ISSUE TITLE
Import from VB does not report Bad Reference

### ISSUE THREAD

**Issue by dmrvb** [twinbasic/twinbasic#1427]
**Describe the bug**
Importing a project from VB with a defective reference (file not present because I have moved it) gives no warning of failure in the  import log.
A success message is given if the reference is correct.

**To Reproduce**
Make a VB project with a reference, move/remove the referenced file, import into tB

**Expected behavior**
Other import failures are reported so this one should be too.

**Additional context**

The bad reference in the VBP file is:
Reference=*\G{F9015E81-CAAC-45C0-94E8-42B7DA5D7557}#4.2a#0#C:\Windows\SysWOW64\oleexp.tlb#OLEEXP - olelib With Modern Interfaces by fafalone, v4.42

**Comment by WaynePhillipsEA** [comment]
Currently, the import routine works by simply migrating the VBP into the twinproj file structure.  No checks are made as to the availability of external components or type libraries.

Errors such as missing type libraries and ActiveX controls need to be reported when opening a twinproj project anyway, so it makes sense not to duplicate that work during migration.

However, missing type libraries/ActiveX components are not being reported particularly well upon opening a project, currently.  If you look closely, there might be a warning in the Debug Console when you open a project, but this needs improving / replacing with a prominent modal warning dialog.

**Comment by dmrvb** [comment]
OK, This should not be high priority as we really should not be importing defective VB6 code into tB.
Note that if I am foolish enough to try to run this code desoite all the reported problems then I get this 
![image](https://user-images.githubusercontent.com/83877219/208304628-b3964e5c-4890-4df8-a68e-41858a483042.png)

**Comment by WaynePhillipsEA** [comment]
Yeah, if there are compilation errors, the debugger will halt at the procedure start, like the VBx debugger.

**Comment by fafalone** [comment]
Assuming the file is there it shouldn't fail to import at all. Is it a version thing? VB is very inconsistent with whether it makes you re-do the version reference... and tB is weird in listing multiple different versions all pointing at the same path, with the same GUID... VB only lists the one, though it often has the wrong version number shown in the references dialog. Sometimes it's the same GUID and same version and VB fails to load it and you have to reimport anyway... though this is mostly when opening a project on a new Windows install, even though that shouldn't make a difference if it's registered.

(PS- oleexp won't work in 64bit at all; I recommend instead of importing that to instead click 'twinpack packages' under the reference list and select twinBASIC Shell Library, which is the 64bit-compatible successor to oleexp. It has all the interfaces and APIs declared the same way, including Core Audio, and all the addon modules like mCoreAudio built in. It's on the package server, which I do _very carefully_ as the VSCode extension reports thousands of errors and tries to autocorrect some 'wrong' syntax... but it works)

**Comment by dmrvb** [comment]
Thanks
It was my fault, I had upgraded to a newer version of your oleexp and put it into a different folder, but had not corrected the VBP, so it was inevitable that tB could not find it. I was just surprised not to see an error message.
My code won't even run in 32 bit mode yet, though I suspect its close. I am very keen to go to be 64bit but thats a way off, and I have noted that you are working on a 64bit oleexp equivalent for tB. Many thanks, as your oleexp has been essential to my software. I assume I cant do 64bit till Wayne makes all the controls 64bit compatible?
Also I am still very interested in your ucShellBrowse project...are you going to get that sorted in tB?

**Comment by fafalone** [comment]
All of the controls present so far in tB are 64bit compatible; I've already converted some fairly complex applications to 64bit like VBEventTrace. For the common controls I've found if you have MS Office 64bit installed, you can use the 64bit ComCtl ocx that are included with it. I've also my tbComCtlLib project that has 64bit-compatible definitions of all ComCtl APIs/types/etc, if you wanted to make API-based controls like in my event tracer.

I've been working on getting ucShellBrowse going now that there's UC support, but as UC support is still brand new and it's a very, very complex control, there's still a lot of issues preventing it from working correctly... but it's getting closer as Wayne fixes the bugs (or gets his time wasted because it's a bug of mine lol)... it can be placed and started up to the initial folder (in both 32bit and 64bit mode), but doing much beyond that either crashes or doesn't work.

[ucShellTree](https://github.com/fafalone/ucShellTree) on the other hand is nearly fully working in both 32bit and 64bit mode, you just need to make sure you have a Form_Resize event in the host form that resizes it based on Form size (even if you don't want it resizable, you'd currently need to do something to change it by a pixel or two, constant sizing won't work). There's a demo form in the project.

If interested you can also check out [my other repositories](https://github.com/fafalone) (I just have twinBASIC stuff on this account), I've got a number of other examples of my projects ported to 64bit compatible, including a couple more using shell interfaces found in oleexp/tbShellLib.
